### PR TITLE
chore: Change all the references of OwlBot.yaml to OwlBot-hermetic.yaml.

### PR DESCRIPTION
**Pull request by blakeli0**
This is follow up of https://github.com/googleapis/google-cloud-java/pull/10696. The hermetic build generation process would fail without this PR due to unable to find OwlBot.yaml files. 
This PR changes all the references of `OwlBot.yaml` to `OwlBot-hermetic.yaml`, removes `OwlBot.yaml` from a few places that don't need to reference it in the first places.

### CODE REVIEW

File: library_generation/postprocess_library.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ for required_input in "${required_inputs[@]}"; do
   fi
 done
 
-for owlbot_file in ".repo-metadata.json" "owlbot.py" ".OwlBot.yaml"
+for owlbot_file in ".repo-metadata.json" "owlbot.py" ".OwlBot-hermetic.yaml"
```
Comment (JoeWang1127) [inline library_generation/postprocess_library.sh:47]: Could you extract these values to variables?

Comment (blakeli0) [reply library_generation/postprocess_library.sh:47]: For String literals, it is usually preferred to use it directly as it is more readable, especially if it is only used once. I do see `.OwlBot.yaml` being used multiple times, so I'll extract it and leave the other two as they are.

**Review by JoeWang1127 — APPROVED** [review]

**Review by diegomarquezp — APPROVED** [review]
LGTM, thank you

File: library_generation/postprocess_library.sh [hunk 2]
```diff
@@ -74,12 +75,12 @@ if [[ "${is_monorepo}" == "true" ]]; then
   # - "/google-.*/src"
 
   library_name=$(basename "${postprocessing_target}")
-  cat "${postprocessing_target}/.OwlBot.yaml" \
+  cat "${postprocessing_target}/${owlbot_yaml_file_name}" \
     | sed "s/- \"\/${library_name}/ - \"/" \
     > "${postprocessing_target}/.OwlBot.hermetic.yaml"
```
[the remark this answers is not in the document]
Comment (blakeli0) [reply library_generation/postprocess_library.sh:80]: Agreed!

**Pull request merged**